### ISSUE TITLE
Does f-stack support keepalived?

### ISSUE THREAD

**Issue by chenjianjian** [F-Stack/f-stack#457]
Does f-stack support keepalived?

**Comment by jfb8856606** [comment]
No, F-Stack does not support keepalived out of the box. F-Stack only ports a few applications (nginx, redis-server, etc.) to its user-space API, and keepalived has not been ported.

More fundamentally, keepalived relies on **kernel networking** — VRRP multicast, kernel routing/IP management, Netlink, and ipvs (LVS). F-Stack is a user-space stack that owns the NIC via DPDK, so those kernel mechanisms don't exist in the F-Stack data path.

Practical options:
1. Run keepalived in the **kernel** (bound to F-Stack's KNI interface or a separate management NIC) purely for **VIP failover / health checks**, while the F-Stack application handles the data plane — i.e. keep the two separated.
2. Implement HA/LB at the F-Stack layer yourself, or use a DPDK-native load-balancing solution.

Closing as answered — feel free to reopen with a specific deployment scenario.
